Userland/Utilities/test-fuzz.cpp

```diff
@@ -0,0 +1,130 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/String.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#define ENUMERATE_TARGETS(T) \
+    T(BMPLoader)             \
+    T(ELF)                   \
+    T(Gemini)                \
+    T(GIFLoader)             \
+    T(HttpRequest)           \
+    T(ICOLoader)
+/*
+//    T(JPGLoader) \
+//    T(Js) \
+//    T(Markdown) \
+//    T(PBMLoader) \
+//    T(PGMLoader) \
+//    T(PNGLoader) \
+//    T(PPMLoader) \
+//    T(RegexECMA262) \
+//    T(RegexPosixExtended) \
+//    T(Shell) \
+//    T(TTF) \
+//    T(URL)
+*/
+
+#undef __ENUMERATE_TARGET
+#define __ENUMERATE_TARGET(x) extern "C" int Test##x(const uint8_t*, size_t);
+ENUMERATE_TARGETS(__ENUMERATE_TARGET)
+#undef __ENUMERATE_TARGET
+
+// FIXME: How to automate this?
+#define LLVMFuzzerTestOneInput TestBMPLoader
+#include <Meta/Lagom/Fuzzers/FuzzBMPLoader.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestELF
+#include <Meta/Lagom/Fuzzers/FuzzELF.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestGemini
+#include <Meta/Lagom/Fuzzers/FuzzGemini.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestGIFLoader
+#include <Meta/Lagom/Fuzzers/FuzzGIFLoader.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestHttpRequest
+#include <Meta/Lagom/Fuzzers/FuzzHttpRequest.cpp>
+#undef LLVMFuzzerTestOneInput
+
+#define LLVMFuzzerTestOneInput TestICOLoader
+#include <Meta/Lagom/Fuzzers/FuzzICOLoader.cpp>
+#undef LLVMFuzzerTestOneInput
+
+typedef int (*TestOneInputFn)(const uint8_t*, size_t);
+
+static TestOneInputFn parse_target_name(const String& name)
+{
+    if (name == "list") {
+        outln("The following fuzzers are included:");
+#undef __ENUMERATE_TARGET
+#define __ENUMERATE_TARGET(x) outln(#x);
+        ENUMERATE_TARGETS(__ENUMERATE_TARGET)
+#undef __ENUMERATE_TARGET
+        exit(0);
+    }
+
+#undef __ENUMERATE_TARGET
+#define __ENUMERATE_TARGET(x) \
+    if (name == #x)           \
+        return Test##x;
+    ENUMERATE_TARGETS(__ENUMERATE_TARGET)
+#undef __ENUMERATE_TARGET
+
+    warnln("Unknown fuzzing target \"{}\". Try \"list\" to get a full list.", name);
+    exit(1);
+}
+
+int main(int argc, char** argv)
+{
+    const char* type;
+    const char* filename;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(type, "Which type of fuzzer you'd like to try (use \"list\" to list all existing)", "fuzz-kind");
+    args_parser.add_positional_argument(filename, "Which file to use as input", "filename");
```

Comment:
I think it's more common to write something like "Type of fuzzer to run" and "Fuzzer input file" instead?
Just a nit